### PR TITLE
Implement support for key revision.tag=last

### LINKED ISSUE #23: Implement support for specifying "latest" on tag block to attempt to resolve it from list of possible tags

### PR DESCRIPTION
**Pull request by Grokzen**
Closes #23 

Implement support for specifying "revision.tag=latest" key and support for a few helper options that can be used to filter and clean tags to allow for semver to work it's magic. Please see docs for additional examples and usage of all options.

### CODE REVIEW

File: docs/revision-tag.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      # Pre-processes the tags before matching and removes any that we do not want
+      # This step runs before clean step.
+      # Multiple regex values is supported but it will stop parsing at first match and remove it
+      tag-filter-regex:
+        - "([0-9].[0-9].[0-9])"
+
+      # Pre-processes the tags before selection step. Create a regex with a group that you want
+      # to extract the semver data from. Multiple regex values is supported but first value matching
+      # will be used before moving on to next tag.
+      # This step is usefull to remove and prefix or suffixes from tags and only care about the semver part when comparing tags
```
Comment (holmboe) [inline docs/revision-tag.md:23]:
```suggestion
      # This step is useful to remove any prefix or suffix from tags, only affecting the semver part when comparing tags
```

File: sgit/core.py [hunk 2]
```diff
@@ -3,14 +3,16 @@
 # python std lib
 import logging
 import os
+import re
 import sys
 
 # sgit imports
-from sgit.exceptions import SgitException, SgitConfigException
+from sgit.exceptions import SgitException, SgitConfigException, SgitConfigException
```
Comment (holmboe) [inline sgit/core.py:10]: `SgitConfigException` twice?

File: sgit/enums.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+class OrderAlgorithms(Enum):
+    ALPHABETICAL = 10
+    TIME = 20
+    SEMVER = 30
+
+
+class SelectionMethods(Enum):
+    SEMVER = 10
+    EXACT = 20
```
Comment (holmboe) [inline sgit/enums.py:12]: Maybe use `LITERAL` instead of `EXACT`? You decide.

**Review by holmboe — APPROVED** [review]
Minor comment. LGTM.

File: sgit/core.py [hunk 4]
```diff
@@ -3,23 +3,28 @@
...
 # sgit imports
-from sgit.exceptions import SgitException, SgitConfigException
+from sgit.constants import *
+from sgit.enums import *
+from sgit.exceptions import *
 
 # 3rd party imports
-import ruamel
 import git
+import semver
```
Comment (holmboe) [inline sgit/core.py:16]: Unused import.

File: sgit/enums.py (hunk 3 above)
Comment (Grokzen) [reply sgit/enums.py:12]: I might, but let me think about it and i will push it in a future commit if i change it. I find both options a little bit wonky tbh

Comment (Grokzen) [reply sgit/enums.py:12]: I added in this as a idea card in the 1.0.0 project for keeping track of this suggestion

**Pull request merged**